ioSPI/fourier.py

```diff
@@ -0,0 +1,212 @@
+import numpy as np
+import pyfftw.interfaces.numpy_fft
+from numba import jit
+
+
+@jit
+def neg_pos_2d(arr2d):
+    """Apply checkerboard pattern.
+
+    Each pixel switches from positive to negative in
+    checker board pattern.
+
+    Parameters
+    ----------
+    arr2d : numpy.ndarray, shape (N,N)
+        Input array.
+
+    Returns
+    -------
+    arr2d : numpy.ndarray, shape (N,N)
+        Output array.
+    """
+    assert arr2d.ndim == 3  # extra axis
+    for n_particle in range(arr2d.shape[0]):
+        for r in range(arr2d.shape[1]):
+            for c in range(arr2d.shape[2]):
+                if (r + c) % 2:
+                    arr2d[n_particle, r, c] *= -1
+    return arr2d
+
+
+@jit
+def make_neg_pos_3d(arr3d):
```

Comment:
I am not sure it is very clear from the function name what distinguishes this function from the previous one.

Reply:
One is for 2D FFT, the other is for 3D FFT. They have different criteria for the checkerboard:
`if (r + c) % 2:`
vs
`if (r1 + r2 + r3) % 2:`

Reply:
yes but one is `make_` and the other is not, they are not used the same way which might be confusing?

Reply:
Oh that is a good point. I didn't pick up on that. One was doing things in place, and another way outputting the checkerboard array. I'll make them both output a checkerboard array. The tests will make sure everything is working and we get the same array back after doing the iFT of the FT